lib/oauth.js

```diff
@@ -558,7 +558,7 @@ internals.encode = function (string) {
 internals.parse = function (payload) {
 
     payload = Buffer.isBuffer(payload) ? payload.toString() : payload;
-    if (payload[0] === '{') {
+    if (/^\s*{/.test(payload)) {
```

Comment:
You could technically be clobbering Querystring params and/or values with whitespace if the test failed and it fell through to `Querystring.parse`.  Obviously that's super edge case and I've never personally seen query params with whitespace in their names, but that's anecdotal and this felt safer.